src/util/Helpers.hpp

```diff
@@ -37,4 +37,26 @@ QColor getRandomColor(const QString &userId);
 QString formatUserMention(const QString &userName, bool isFirstWord,
                           bool mentionUsersWithComma);
 
+template <typename T>
+std::vector<T> splitListIntoBatches(const T list, int batchSize = 100)
+{
+    std::vector<T> batches;
```

Comment:
```suggestion
    int batchCount = (list.size() / batchSize) + 1;
    std::vector<T> batches;
    batches.reserve(batchCount);
```

Reply:
@iProdigy instead of just giving out suggestion please tell me why and/or what's the benefit here. I had that call to `reserve` function in the previous implementation but I couldn't find any reference or documentation on whether I can/could/should/have to do this. it doesn't seem to change anything and is just unnecessary imo. I can add this, I just want to know why...

Reply:
Since `std::vector` has a dynamic size, it may repeatedly reallocate space as elements are added to ensure there is sufficient space. These allocations are not costless & can be optimized by simply indicating the target size

Reply:
Thank you so much! It's very useful to know for the future, I'll add this fix right now.